sys/malloc.c

```diff
@@ -9,43 +9,36 @@ extern uint8_t __ebss[];
 static struct {
   void *ptr; /* Pointer to the end of kernel's bss. */
   void *end; /* Limit for the end of kernel's bss. */
-  mtx_t lock;
   bool shutdown;
-} sbrk = {__ebss, __ebss + 512 * PAGESIZE, MTX_INITIALIZER, false};
+} sbrk = {__ebss, __ebss + 512 * PAGESIZE, false};
 
 void kernel_brk(void *addr) {
   if (sbrk.shutdown)
     panic("Trying to use kernel_brk after it's been shutdown!");
-  mtx_lock(sbrk.lock);
```

Comment:
I agree that current `mtx_lock` implementation should be removed. However leaving `sbrk` calls without synchronization seems to be light-minded. Is it possible to replace `mtx_*` with critical section primitive in this case?